generate/genhcl/genhcl.go

```diff
@@ -147,6 +149,36 @@ func Load(
 	var hcls []HCL
 	for _, hclBlock := range hclBlocks {
 		name := hclBlock.Label
+
+		matchedAnyStackFilter := len(hclBlock.StackFilters) == 0
+		for _, cond := range hclBlock.StackFilters {
+			matched := true
+
+			for n, g := range map[string]glob.Glob{
+				"project path":    cond.ProjectPaths,
+				"repository path": cond.RepositoryPaths,
+			} {
+				if g != nil {
```

Comment:
```suggestion
				if g != nil && !g.Match(st.Dir.String()) {
```
you can inline the ifs.